jobs/rabbitmq-server/templates/post-start.bash

```diff
@@ -46,60 +46,60 @@ grant_permissions_for_all_vhosts() {
     set -e
 }
 
-create_operator_admin() {
-  if [ -n "$RMQ_OPERATOR_USERNAME" ]
-  then
-    echo "$RMQ_OPERATOR_USERNAME" > $OPERATOR_USERNAME_FILE
-    create_admin "$RMQ_OPERATOR_USERNAME" "$RMQ_OPERATOR_PASSWORD"
-  fi
-}
+create_admin() {
+  username_file=$1
+  username=$2
+  password=$3
 
-create_broker_admin() {
-  if [ -n "$RMQ_BROKER_USERNAME" ]
+  if [ -z "$username" ]
   then
-    echo "$RMQ_BROKER_USERNAME" > $BROKER_USERNAME_FILE
-    create_admin "$RMQ_BROKER_USERNAME" "$RMQ_BROKER_PASSWORD"
+    return
   fi
-}
-
-delete_operator_admin() {
-  set +e
-  USERNAME=$(cat $OPERATOR_USERNAME_FILE)
-  "${RMQ_CTL}" delete_user "$USERNAME" 2>&1
-  set -e
-  true
-}
 
-delete_broker_admin() {
+  echo "$username" > "$username_file"
   set +e
```

Comment:
Rather turning shell error handling off then back on again:

```bash
set +e
command-that-might-fail-but-ok
set -e
```

You can also do:

```bash
command-that-might-fail-but-ok || true # ok if fails
```

Reply:
True! Thanks a lot for reviewing :)